Classes/Common/DocumentList.php

```diff
@@ -316,6 +252,116 @@ protected function getRecord($element)
         return $record;
     }
 
+    /**
+     * It gets SOLR result
+     *
+     * @access private
+     *
+     * @param array $record: for searched document
+     *
+     * @return Result
+     */
+    private function getSolrResult($record) {
+        $fields = Solr::getFields();
+
+        $query = $this->solr->service->createSelect();
+        // Restrict the fields to the required ones
+        $query->setFields($fields['uid'] .',' . $fields['id'] .',' . $fields['toplevel'] .',' . $fields['thumbnail'] .',' . $fields['page']);
+        foreach ($this->solrConfig as $solr_name) {
+            $query->addField($solr_name);
+        }
+        // Set additional query parameters.
+        // Set reasonable limit for safety reasons.
+        // We don't expect to get more than 10.000 hits per UID.
+        $query->setStart(0)->setRows(10000);
+        // Take over existing filter queries.
+        $filterQueries = isset($this->metadata['options']['params']['filterquery']) ? $this->metadata['options']['params']['filterquery'] : [];
+        // Extend filter query to get all documents with the same UID.
+        foreach ($filterQueries as $key => $value) {
+            if (isset($value['query'])) {
+                $filterQuery[$key] = $value['query'] . ' OR ' . $fields['toplevel'] . ':true';
```

Comment:
```suggestion
                $filterQuery[$key] = $value['query'] . ' OR ' . $fields['toplevel'] . ':true';
                $filterQuery = [
                    'key' => $key,
                    'query' => $value['query'] . ' OR ' . $fields['toplevel'] . ':true'
                ];
```

Solarium expects an array like this at this point. Important is the key "key". The key "query" seems not to matter.

The other approach would be in the above foreach loop:

`$query->createFilterQuery('kitodo' . $key)->setQuery($value['query'] . ' OR ' . $fields['toplevel'] . ':true');`

This calls addFilterQuery internally.

Reply:
Oh, now I understand. Thanks!